core/app/c/[communitySlug]/pubs/[pubId]/page.tsx

```diff
@@ -127,6 +126,19 @@ export default async function Page({
 		}
 	);
 
+	const canView =
+		(await userCan(Capabilities.viewPub, { type: MembershipType.pub, pubId }, user.id)) ||
+		(pub.stageId &&
+			(await userCan(
+				Capabilities.viewStage,
+				{ type: MembershipType.stage, stageId: pub.stageId },
+				user.id
+			)));
```

Comment:
Unfortunately we have to do this check _after_ fetching the pub, because we need its stage too

Reply:
ah right, that's very annoying. But can't we skip the `viewPub` check, as that's done in `getPubs` anyway? I guess it then becomes hard to say whether or not we don't have a pub bc you aren't allowed to view it or bc the pub does not exist.

Reply:
you could theoretically also move the Stage display to a separate component that's `<Suspended>` and only do the stage check there

but actually, i think we should just let you view the name of the stage on this page, since stage names are kind of like statuses.

Reply:
if you're a contributor, i think you should be able to see the pub page of your article and which stage it's in, even if you cannot see the stage completely

Reply:
i suppose you're right about skipping the viewPub check. i'm guessing it throws an error when no pub is returned anyways.

i'm not sure i follow about the stage check though. i'm not checking that capability here for displaying the name of the stage. i'm checking it and combining here with `||` because having `viewStage` is sufficient for viewing the pub even if you don't have `viewPub` directly on the pub

Reply:
and now that i've said that, i'm remembering that i specifically implemented that logic inside of usercan 🤦

Reply:
so i may not need either of these really, but i will put the viewPub one before the pub fetching

Reply:
ah nevermind, that makes sense!

but yeah, i think still having the `userCan` check to be able to redirect users is better than just showing "Pub Not Found"